### PR TITLE
Add organization discovery for b2b login flow

### PR DESCRIPTION
**Pull request by dewniMW**
Issue https://github.com/wso2/product-is/issues/16379
BE changes will be added with https://github.com/wso2-extensions/identity-auth-organization-login/pull/30

UI when organization discovery is enabled:
Screen Shot 2023-10-15 at 5 06 44 PM

Error displayed when unable to identify organization name from the provided email domain:
Screen Shot 2023-10-15 at 5 08 19 PM

UI to enter organization name instead of going with discovery:
Screen Shot 2023-10-12 at 3 46 17 PM

UI when organization discovery is disabled:
Screen Shot 2023-10-12 at 4 11 20 PM

### CODE REVIEW

File: identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_de_DE.properties [hunk 1]
```diff
@@ -446,6 +446,11 @@ error.504=Fehler 504 - Timeout von Gateway
 # Organization Login
 organization.login=Organisation Login
 organization.name=Name der Organisation
```
Comment (Achintha444) [inline identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_de_DE.properties:448]: Need to replace this value as well

File: identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_ja_JP.properties [hunk 2]
```diff
@@ -458,7 +458,13 @@ error.504=エラー 504 - ゲートウェイのタイムアウト
 
 ## Organization Login
 organization.login=組織ログイン
-organization.name=組織の名前
+organization.name=組織名
```
Comment (Achintha444) [inline identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_ja_JP.properties:461]: Seems like there is an error when translating to Japanesee

File: identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_de_DE.properties (hunk 1 above)
Comment (dewniMW) [reply identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_de_DE.properties:448]: this is the correct translation

File: identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_ja_JP.properties (hunk 2 above)
Comment (dewniMW) [reply identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_ja_JP.properties:461]: this is the correct translation given in google translate

File: identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_de_DE.properties (hunk 1 above)
Comment (Achintha444) [reply identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_de_DE.properties:448]: Ack

File: identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_ja_JP.properties (hunk 2 above)
Comment (Achintha444) [reply identity-apps-core/apps/authentication-portal/src/main/resources/org/wso2/carbon/identity/application/authentication/endpoint/i18n/Resources_ja_JP.properties:461]: Ack

File: identity-apps-core/apps/authentication-portal/src/main/webapp/org_discovery.jsp [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+      %>
+      <jsp:include page="extensions/header.jsp"/>
+      <%
+         } else {
+      %>
+      <jsp:include page="includes/header.jsp"/>
+      <%
+         }
+      %>
+      <!--[if lt IE 9]>
```
Comment (dasuni-30) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/org_discovery.jsp:73]: Shall we remove the redundant comment?

File: identity-apps-core/apps/authentication-portal/src/main/webapp/org_name.jsp [hunk 4]
```diff
@@ -120,17 +121,31 @@
...
-                            </button>
-                        </div>
+                        <input type="submit" id="submitButton" onclick="submitOrgName(); return false;"
+                            value="<%=AuthenticationEndpointUtil.i18n(resourceBundle, "submit")%>"
+                            class="ui primary large fluid button" />
+                        <% if (isOrgDiscoveryEnabled) { %>
+                            <div class="ui horizontal divider"><%=AuthenticationEndpointUtil.i18n(resourceBundle, "or")%></div>
+                            <div class="social-login blurring social-dimmer">
+                                <input type="submit" id="discoveryButton" onclick="promptDiscovery();" class="ui fluid button"
+                                    value="Provide email address">
```
Comment (dasuni-30) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/org_name.jsp:146]: Can we add string localization here?

**Review by dasuni-30 — APPROVED** [review]

**Review by Achintha444 — APPROVED** [review]

**Pull request merged**